### PR TITLE
feat: Add exact flag to InstCombine's EDSL

### LINKED ISSUE #730: sdiv exact keyword not preserved
It seems the exact keyword in the generated proof is not preserved: https://github.com/opencompl/lean-mlir/blob/6e76106791a32338882c53685cd860527adda64d/SSA/Projects/InstCombine/tests/LLVM/gexact.lean#L20

The original test case for this is:

```llvm
define i32 @sdiv2(i32 %x) {
; CHECK-LABEL: @sdiv2(
; CHECK-NEXT:    [[Y:%.*]] = ashr exact i32 [[X:%.*]], 3
; CHECK-NEXT:    ret i32 [[Y]]
;
  %y = sdiv exact i32 %x, 8
  ret i32 %y
}
```

Our incomplete semantics currently lead to verification failures in the test data @luisacicolini generates. @lfrenot, can you look into this on Monday?

### PR DESCRIPTION
**Pull request by lfrenot**

### CODE REVIEW

File: SSA/Projects/InstCombine/LLVM/EDSL.lean [hunk 1]
```diff
@@ -110,12 +110,28 @@ def mkExpr (Γ : Ctxt (MetaLLVM φ).Ty) (opStx : MLIR.AST.Op φ) :
...
-        | "llvm.ashr"   => pure <| Sum.inl .ashr
         | "llvm.urem"   => pure <| Sum.inl .urem
         | "llvm.srem"   => pure <| Sum.inl .srem
-        | "llvm.sdiv"   => pure <| Sum.inl .sdiv
-        | "llvm.udiv"   => pure <| Sum.inl .udiv
+        | "llvm.lshr"   => do
+          let attr? := opStx.attrs.getAttr "isExact"
+          match attr? with
+            | .none => pure <| Sum.inl .lshr
+            | .some _ => pure <| Sum.inl (MOp.BinaryOp.lshr ⟨true⟩)
```
Comment (bollu) [inline SSA/Projects/InstCombine/LLVM/EDSL.lean]:
```suggestion
          let isExact? := opStx.attrs.getAttr "isExact"
          pure <| Sum.inl (.lshr ⟨isExact?.isSome⟩)
```

Let's do it this way, it reduces the amount of duplication by quite a bit!

**Review by bollu — COMMENTED** [review]
Otherwise LGTM

**Review by tobiasgrosser — APPROVED** [review]
Cool. Let's then merge this after the change request got addressed.

File: SSA/Projects/InstCombine/LLVM/EDSL.lean (hunk 1 above)
Comment (lfrenot) [reply SSA/Projects/InstCombine/LLVM/EDSL.lean]: Done

**Pull request merged**